openshift/console
# Serverless function creation with Pipelines

Story: https://issues.redhat.com/browse/ODC-7235

Screenshots/GIF:
- Create Serverless function via Import from Git with Pipeline
![Create-serverless-function-via-Import-from-Git-with-Pipeline](https://user-images.githubusercontent.com/2561818/217762533-4a96fef2-03a8-48d4-a53b-3280af706467.gif)

- Create Serverless function via Create Serverless form with Pipeline
![Create-serverless-function-via-Create-Serverless-form-with-Pipeline gif](https://user-images.githubusercontent.com/2561818/217763372-d2c2ab37-b0ea-49ca-82a9-c60624576cf7.gif)

Test setup:
- Install Pipeline and Serverless Operator and create CRs
- Add all ClusterTasks and Pipeline from this gist https://gist.github.com/vikram-raj/386b4eca5193036b0ca29471f6e513a5
- Navigate to Import from Git form and enter Git URL https://github.com/vikram-raj/hello-func-node
- Check the Pipeline checkbox and click on create
- Check in the Topology that the Serverless function is get created and a Pipeline decorator is also attached to it.
- Also the Create Serverless function form by entering Git URL https://github.com/vikram-raj/hello-func-node and enable Pipeline in the form
